### PR TITLE
docs(chapter-5): add cargo.md

### PR DESCRIPTION
**Pull request by r17x**

### CODE REVIEW

File: content/capter-5/cargo.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+* **_Paths_** adalah sebuah file yang terisolasi untuk mengelompokan _function_ dan _module_. 
+
+
+### _Packages & Crates_
+
+_Crates_ adalah kumpulan _modules_, dan _function_ yang saling berelasi di dalam sebuah _package_.
+
+
+```
```
Comment (adiatma) [inline content/capter-5/cargo.md]:
```bash
cargo new my-project
Created binary (application) `my-project` package
ls my-project
Cargo.toml
src
ls my-project/src
main.rs
```

File: content/capter-5/cargo.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+### _Modules_
+
+Setelah membahas tentang _package_ dan _crate_ hal selanjutnya adalah _modules,_ kita akan bahas sedikit tentang bagian - bagian dari sebuah _module_. Hal yang pertama adalah _paths_ yang digunakan untuk mengelompokan sebuah fungsi - fungsi dan module dalam sebuah file. Selanjutnya ada **_use_** yang merupakan _keyword_ untuk memanggil atau membawa sebuah _path_ kedalam _scope_ yang berbeda, dan **_pub keyword_** yang akan membuat sebuah _item_ menjadi publik agar dapat diakses di _scope_ yang berbeda.
+
+_Modules_ membuat kita mudah untuk mengorganisasikan sebuah _function_ berdasarkan fungsinya dan dapat digunakan berkali - kali untuk meminimalisir pekerjaan kita.
+
+Oke, untuk selanjutnya kita akan membuat sebuah paket _library_ di Rust, silahkan ketik perintah `cargo new --lib my-lib` yang secara otomatis akan membuat skeleton project yang _support_ untuk menerapkan konsep _library_, dengan _option_ `--lib` file yang di _generate_ dalam direktori `src` adalah `src/lib.rs` berbeda dengan tanpa menggunakan `--lib`.
+
+_src/lib.rs_
```
Comment (adiatma) [inline content/capter-5/cargo.md]:
```rust
// src/lib.rs
pub mod restaurant {
   pub fn hello(name: String) -> String {
       name
   }
}
```

File: content/capter-5/cargo.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+       name
+   }
+}
+```
+
+Pada contoh kode diatas terdapat sebuah _module_ restaurant dengan _function_ `hello()` di dalamnya. _Keyword_ **_pub_** digunakan untuk membuat _module_ tersebut bisa diakses dari scope atau _path_ yang berbeda.
+
+Untuk menguji _module_ restaurant tersebut silahkan buat file `main.rs`, di dalam direktori `src/main.rs` isi dengan kode dibawah ini.
+
+_src/main.rs_
```
Comment (adiatma) [inline content/capter-5/cargo.md]:
```rust
// src/main.rs
extern crate my_lib;
use my_lib::restaurant;
fn main() {
   let adi = restaurant::hello(String::from("adi"));
   println!("{}", adi);
}
```

File: content/capter-5/cargo.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+### _Paths_ untuk mereferensi ke sebuah item di dalam _module_
+
+Ada dua _path_ yang terkenal di Rust, yaitu, **_absolute path_** & **_relative path,_** berikut penjelasannya.
+
+*   **_Absolute path_** diawali dengan _keyword crate_ untuk mengambil modul dari _root component_.
+*   **_Relative path_** menggunakan _self_ dan _super_ untuk mengambil dari _current module_.
+
+Dari kedua _absolute_ dan _relative_ _paths_ diatas selanjutnya untuk mengakses module secara spesifik bisa menggunakan semicolons `(::)` biar lebih `clear` berikut ada contohnya.
+
+> _src/lib.rs_
```
Comment (adiatma) [inline content/capter-5/cargo.md]:
```rust
// src/lib.rs
pub mod restaurant {
   pub fn hello(name: String) -> String {
       name
   }
}
pub fn call_directly() {
   // Absolute Path
   crate::restaurant::hello();
   // Relative Path
   self::restaurant::hello();
}
```

File: content/capter-5/cargo.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+   // Relative Path
+   self::restaurant::hello();
+}
+```
+
+
+_Keyword_ **_pub_** yang terdapat di awal _module_ dan _function_ bertujuan untuk mengekspos _module_ dan _function_ agar bisa diakses di luar _scope_, karena secara default module dan function di Rust bersifat _private_.
+
+
+###
```
Comment (adiatma) [inline content/capter-5/cargo.md]: remove `###`

File: content/capter-5/cargo.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+
+_Keyword_ **_pub_** yang terdapat di awal _module_ dan _function_ bertujuan untuk mengekspos _module_ dan _function_ agar bisa diakses di luar _scope_, karena secara default module dan function di Rust bersifat _private_.
+
+
+### 
+
+
+### Memahami **_use_**
+```
```
Comment (adiatma) [inline content/capter-5/cargo.md]:
```rust
pub mod restaurant {
   pub fn hello(name: String) -> String {
       name
   }
}

use crate::restaurant;

pub fn call_hello_from_restaurant() {
   restaurant::hello()
}
```

**Review by adiatma — CHANGES_REQUESTED** [review]

File: content/capter-5/cargo.md (hunk 6 above)
Comment (r17x) [reply content/capter-5/cargo.md]: updated

File: content/capter-5/cargo.md (hunk 5 above)
Comment (r17x) [reply content/capter-5/cargo.md]: updated

File: content/capter-5/cargo.md (hunk 4 above)
Comment (r17x) [reply content/capter-5/cargo.md]: updated

**Comment by adiatma** [comment]
Format `##` kaga ke baca kayaknya pak @ri7nz  di markdownnya, makanya kemarin ku hapus wkwk

**Comment by r17x** [comment]
@adiatma  `##` format H2 kan ya, ngk kebaca diweb atau dimana nih ?

**Comment by adiatma** [comment]
> @adiatma `##` format H2 kan ya, ngk kebaca diweb atau dimana nih ?

Iya `h2` gak kebaca di web

**Comment by r17x** [comment]
@adiatma 

bisa harusnya 

https://handbook.evilfactory.id/evangelist/2-voice-and-tone
https://github.com/evilfactorylabs/handbook/blob/master/content/evangelist/2-voice-and-tone.md#perbedaan-suara--nada

**Comment by adiatma** [comment]
> @adiatma
> 
> bisa harusnya
> 
> https://handbook.evilfactory.id/evangelist/2-voice-and-tone
> https://github.com/evilfactorylabs/handbook/blob/master/content/evangelist/2-voice-and-tone.md#perbedaan-suara--nada

Hmm, iya ya, wait, ku coba lagi Pak..

**Comment by r17x** [comment]
Need review @adiatma  @kevanantha

File: content/chapter-5/cargo.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+---
+## Chapter V - Cargo
+
+### _Mengenal Packages, Crates, & Module_
+
+Dalam _development_ software tentunya bertambahnya fitur akan seiring dengan basis kode yang semakin besar, untuk itu Rust punya beberapa konsep yang memungkinkan untuk membuat sebuah komponen yang modular.
+
+* **_Packages_** adalah paket _skeleton_ _project_ yang siap pakai.
+
+* **_Crates_** adalah _root_ dari beberapa module, dan function yang berelasi.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]: italic for module ana function @ri7nz

File: content/chapter-5/cargo.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+* **_Modules_** & **_use _** untuk mengontrol ruang lingkup dan privasi dari _paths _atau bagian dari sistem _module_.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]: `_paths_`

File: content/chapter-5/cargo.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+* **_Paths_** adalah sebuah file yang terisolasi untuk mengelompokan _function_ dan _module_.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]: mengelompokkan

File: content/chapter-5/cargo.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
+---
+title: 'Cargo'
+metaTitle: 'Mengenal Packages, Crates & Modul'
+metaDescription: 'Mengenal Packages, Crates, & Module'
+---
+## Chapter V - Cargo
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion

```

File: content/chapter-5/cargo.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+### _Mengenal Packages, Crates, & Module_
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
_Mengenal Packages, Crates, & Module_
```

File: content/chapter-5/cargo.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+* **_Packages_** adalah paket _skeleton_ _project_ yang siap pakai.
+
+* **_Crates_** adalah _root_ dari beberapa module, dan function yang berelasi.
+
+* **_Modules_** & **_use _** untuk mengontrol ruang lingkup dan privasi dari _paths _atau bagian dari sistem _module_.
+
+* **_Paths_** adalah sebuah file yang terisolasi untuk mengelompokan _function_ dan _module_. 
+
+
+### _Packages & Crates_
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
_Packages & Crates_
```

File: content/chapter-5/cargo.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+
+_Crates_ adalah kumpulan _modules_, dan _function_ yang saling berelasi di dalam sebuah _package_.
+
+
+```
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
```bash
```

File: content/chapter-5/cargo.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+src
+ls my-project/src
+main.rs
+```
+
+
+Pada contoh diatas perintah `cargo new my-project` akan menghasilkan _package_ `my-project` didalamnya terdapat direktori `src` yang berisi file `src/main.rs` dan file `Cargo.toml`  yang berisi informasi paket pustaka yang kita gunakan, mirip seperti `package.json` jika menggunakan _NodeJS_.
+
+
+### _Modules_
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
_Modules_
```

File: content/chapter-5/cargo.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+
+Setelah membahas tentang _package_ dan _crate_ hal selanjutnya adalah _modules,_ kita akan bahas sedikit tentang bagian - bagian dari sebuah _module_. Hal yang pertama adalah _paths_ yang digunakan untuk mengelompokan sebuah fungsi - fungsi dan module dalam sebuah file. Selanjutnya ada **_use_** yang merupakan _keyword_ untuk memanggil atau membawa sebuah _path_ kedalam _scope_ yang berbeda, dan **_pub keyword_** yang akan membuat sebuah _item_ menjadi publik agar dapat diakses di _scope_ yang berbeda.
+
+_Modules_ membuat kita mudah untuk mengorganisasikan sebuah _function_ berdasarkan fungsinya dan dapat digunakan berkali - kali untuk meminimalisir pekerjaan kita.
+
+Oke, untuk selanjutnya kita akan membuat sebuah paket _library_ di Rust, silahkan ketik perintah `cargo new --lib my-lib` yang secara otomatis akan membuat skeleton project yang _support_ untuk menerapkan konsep _library_, dengan _option_ `--lib` file yang di _generate_ dalam direktori `src` adalah `src/lib.rs` berbeda dengan tanpa menggunakan `--lib`.
+
+
+```
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
```rust
```

File: content/chapter-5/cargo.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+   }
+}
+```
+
+Pada contoh kode diatas terdapat sebuah _module_ restaurant dengan _function_ `hello()` di dalamnya. _Keyword_ **_pub_** digunakan untuk membuat _module_ tersebut bisa diakses dari scope atau _path_ yang berbeda.
+
+Untuk menguji _module_ restaurant tersebut silahkan buat file `main.rs`, di dalam direktori `src/main.rs` isi dengan kode dibawah ini.
+
+
+```
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
```rust
```

File: content/chapter-5/cargo.md [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+   let adi = restaurant::hello(String::from("adi"));
+   println!("{}", adi);
+}
+```
+
+
+Pada kode diatas kode `extern crate my_lib` bertujuan untuk mengambil _module_ yang berada di _paths_ `lib.rs` yang merupakan root component, kemudian ada _keyword_ `use` tujuannya untuk membawa _module_ restaurant yang ada pada _package_ `my_lib`.
+
+
+### _Paths_ untuk mereferensi ke sebuah item di dalam _module_
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
_Paths_ untuk mereferensi ke sebuah item di dalam _module_
```

File: content/chapter-5/cargo.md [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+
+Ada dua _path_ yang terkenal di Rust, yaitu, **_absolute path_** & **_relative path,_** berikut penjelasannya.
+
+*   **_Absolute path_** diawali dengan _keyword crate_ untuk mengambil modul dari _root component_.
+*   **_Relative path_** menggunakan _self_ dan _super_ untuk mengambil dari _current module_.
+
+Dari kedua _absolute_ dan _relative_ _paths_ diatas selanjutnya untuk mengakses module secara spesifik bisa menggunakan semicolons `(::)` biar lebih `clear` berikut ada contohnya.
+
+```
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
```rust
```

File: content/chapter-5/cargo.md [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+
+   // Relative Path
+   self::restaurant::hello();
+}
+```
+
+
+_Keyword_ **_pub_** yang terdapat di awal _module_ dan _function_ bertujuan untuk mengekspos _module_ dan _function_ agar bisa diakses di luar _scope_, karena secara default module dan function di Rust bersifat _private_.
+
+### Memahami **_use_**
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
Memahami **_use_**

```

File: content/chapter-5/cargo.md [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+```
```
Comment (adiatma) [inline content/chapter-5/cargo.md]:
```suggestion
```rust
```

File: config.js [hunk 21]
```diff
@@ -21,7 +21,7 @@ const config = {
     }
   },
   sidebar: {
-    forcedNavOrder: ['/introduksi', '/evangelist', '/rnd'],
+    forcedNavOrder: ['/chapter-1', '/chapter-2', '/chapter-3'],
```
Comment (adiatma) [inline config.js]:
```suggestion
    forcedNavOrder: ['/chapter-1', '/chapter-2', '/chapter-3', '/chapter-4', '/chapter-5'],
```

**Review by adiatma — CHANGES_REQUESTED** [review]
cc @ri7nz

File: content/chapter-5/cargo.md [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+Created binary (application) `my-project` package
+ls my-project
+Cargo.toml
+src
+ls my-project/src
+main.rs
+```
+
+
+Pada contoh diatas perintah `cargo new my-project` akan menghasilkan _package_ `my-project` didalamnya terdapat direktori `src` yang berisi file `src/main.rs` dan file `Cargo.toml`  yang berisi informasi paket pustaka yang kita gunakan, mirip seperti `package.json` jika menggunakan _NodeJS_.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
Pada contoh diatas perintah `cargo new my-project` akan menghasilkan _package_ `my-project` didalamnya terdapat direktori `src` yang berisi file `src/main.rs` dan file `Cargo.toml` yang berisi informasi paket pustaka yang kita gunakan, mirip seperti `package.json` jika menggunakan _NodeJS_.
```

File: content/chapter-5/cargo.md [hunk 23]
```diff
@@ -0,0 +1,123 @@
...
+
+
+### _Modules_
+
+Setelah membahas tentang _package_ dan _crate_ hal selanjutnya adalah _modules,_ kita akan bahas sedikit tentang bagian - bagian dari sebuah _module_. Hal yang pertama adalah _paths_ yang digunakan untuk mengelompokan sebuah fungsi - fungsi dan module dalam sebuah file. Selanjutnya ada **_use_** yang merupakan _keyword_ untuk memanggil atau membawa sebuah _path_ kedalam _scope_ yang berbeda, dan **_pub keyword_** yang akan membuat sebuah _item_ menjadi publik agar dapat diakses di _scope_ yang berbeda.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
Setelah membahas tentang _package_ dan _crate_ hal selanjutnya adalah _modules,_ kita akan bahas sedikit tentang bagian-bagian dari sebuah _module_. Hal yang pertama adalah _paths_ yang digunakan untuk mengelompokkan sebuah fungsi-fungsi dan _module_ dalam sebuah file. Selanjutnya ada **_use_** yang merupakan _keyword_ untuk memanggil atau membawa sebuah _path_ kedalam _scope_ yang berbeda, dan **_pub keyword_** yang akan membuat sebuah _item_ menjadi publik agar dapat diakses di _scope_ yang berbeda.
```

File: content/chapter-5/cargo.md [hunk 24]
```diff
@@ -0,0 +1,123 @@
...
+
+_Modules_ membuat kita mudah untuk mengorganisasikan sebuah _function_ berdasarkan fungsinya dan dapat digunakan berkali - kali untuk meminimalisir pekerjaan kita.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
_Modules_ membuat kita mudah untuk mengorganisasikan sebuah _function_ berdasarkan fungsinya dan dapat digunakan berkali-kali untuk meminimalisir pekerjaan kita.
```

File: content/chapter-5/cargo.md [hunk 25]
```diff
@@ -0,0 +1,123 @@
...
+
+Oke, untuk selanjutnya kita akan membuat sebuah paket _library_ di Rust, silahkan ketik perintah `cargo new --lib my-lib` yang secara otomatis akan membuat skeleton project yang _support_ untuk menerapkan konsep _library_, dengan _option_ `--lib` file yang di _generate_ dalam direktori `src` adalah `src/lib.rs` berbeda dengan tanpa menggunakan `--lib`.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
Oke, untuk selanjutnya kita akan membuat sebuah paket _library_ di Rust, silakan ketik perintah `cargo new --lib my-lib` yang secara otomatis akan membuat _skeleton project_ yang _support_ untuk menerapkan konsep _library_, dengan _option_ `--lib` file yang di _generate_ dalam direktori `src` adalah `src/lib.rs` berbeda dengan tanpa menggunakan `--lib`.
```

File: content/chapter-5/cargo.md [hunk 26]
```diff
@@ -0,0 +1,123 @@
...
+```rust
+// src/lib.rs
+pub mod restaurant {
+   pub fn hello(name: String) -> String {
+       name
+   }
+}
+```
+
+Pada contoh kode diatas terdapat sebuah _module_ restaurant dengan _function_ `hello()` di dalamnya. _Keyword_ **_pub_** digunakan untuk membuat _module_ tersebut bisa diakses dari scope atau _path_ yang berbeda.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
Pada contoh kode diatas terdapat sebuah _module_ _**restaurant**_ dengan _function_ `hello()` di dalamnya. _Keyword_ **_pub_** digunakan untuk membuat _module_ tersebut bisa diakses dari _scope_ atau _path_ yang berbeda.
```

File: content/chapter-5/cargo.md [hunk 27]
```diff
@@ -0,0 +1,123 @@
...
+
+Untuk menguji _module_ restaurant tersebut silahkan buat file `main.rs`, di dalam direktori `src/main.rs` isi dengan kode dibawah ini.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
Untuk menguji _module_ _**restaurant**_ tersebut silakan buat file `main.rs`, di dalam direktori `src/main.rs` isi dengan kode dibawah ini.
```

File: content/chapter-5/cargo.md [hunk 28]
```diff
@@ -0,0 +1,123 @@
...
+use my_lib::restaurant;
+
+fn main() {
+   let adi = restaurant::hello(String::from("adi"));
+   println!("{}", adi);
+}
+```
+
+
+Pada kode diatas kode `extern crate my_lib` bertujuan untuk mengambil _module_ yang berada di _paths_ `lib.rs` yang merupakan root component, kemudian ada _keyword_ `use` tujuannya untuk membawa _module_ restaurant yang ada pada _package_ `my_lib`.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
Pada kode diatas, kode `extern crate my_lib` bertujuan untuk mengambil _module_ yang berada di _paths_ `lib.rs` yang merupakan _root component_, kemudian ada _keyword_ `use` tujuannya untuk membawa _module_ _**restaurant**_ yang ada pada _package_ `my_lib`.
```

File: content/chapter-5/cargo.md [hunk 29]
```diff
@@ -0,0 +1,123 @@
...
+
+
+### _Paths_ untuk mereferensi ke sebuah item di dalam _module_
+
+Ada dua _path_ yang terkenal di Rust, yaitu, **_absolute path_** & **_relative path,_** berikut penjelasannya.
+
+*   **_Absolute path_** diawali dengan _keyword crate_ untuk mengambil modul dari _root component_.
+*   **_Relative path_** menggunakan _self_ dan _super_ untuk mengambil dari _current module_.
+
+Dari kedua _absolute_ dan _relative_ _paths_ diatas selanjutnya untuk mengakses module secara spesifik bisa menggunakan semicolons `(::)` biar lebih `clear` berikut ada contohnya.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
Dari kedua _absolute_ dan _relative_ _paths_ diatas selanjutnya untuk mengakses _module_ secara spesifik bisa menggunakan _semicolons_ `(::)` biar lebih `clear` berikut ada contohnya.
```

File: content/chapter-5/cargo.md [hunk 30]
```diff
@@ -0,0 +1,123 @@
...
+   // Absolute Path
+   crate::restaurant::hello();
+
+   // Relative Path
+   self::restaurant::hello();
+}
+```
+
+
+_Keyword_ **_pub_** yang terdapat di awal _module_ dan _function_ bertujuan untuk mengekspos _module_ dan _function_ agar bisa diakses di luar _scope_, karena secara default module dan function di Rust bersifat _private_.
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
_Keyword_ **_pub_** yang terdapat diawal _module_ dan _function_ bertujuan untuk mengekspos _module_ dan _function_ agar bisa diakses di luar _scope_, karena secara _default module_ dan _function_ di Rust bersifat _private_.
```

File: content/chapter-5/cargo.md [hunk 31]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+use crate::restaurant;
+
+pub fn call_hello_from_restaurant() {
+   restaurant::hello()
+}
+```
+
+Pada kode diatas syntax kode `use crate::restaurant;`  digunakan untuk mengambil module restaurant dari **_root module_** dengan menggunakan cara  **_absolute paths._**
```
Comment (kevanantha) [inline content/chapter-5/cargo.md]:
```suggestion
Pada kode diatas, _syntax_ kode `use crate::restaurant;` digunakan untuk mengambil _module restaurant_ dari **_root module_** dengan menggunakan cara **_absolute paths._**
```

**Review by kevanantha — CHANGES_REQUESTED** [review]

**Comment by kevanantha** [comment]
reviewed cc @ri7nz

**Pull request merged**